src/EventListener/AddFormatListener.php

```diff
@@ -145,4 +151,17 @@ private function getNotAcceptableHttpException(string $accept, array $mimeTypes
             implode('", "', $mimeTypes)
         ));
     }
+
+    private function getRequestFormats(Request $request): array
+    {
+        return $this->isDocumentation($request)
+            ? $this->documentationFormats
+            : $this->formats
+        ;
+    }
+
+    private function isDocumentation(Request $request): bool
+    {
+        return 'api_platform.action.documentation' === $request->attributes->get('_controller');
```

Comment:
Alternative proposal:

set a `_api_documentation` attribute here https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Bundle/Resources/config/routing/docs.xml#L10, and test for the presence of this flag. It would be more flexible and give an extension point.

Reply:
@dunglas What you think about a `_api_endpoint_type` set to `documentation` instead ? It may avoid anyone to add another variable later.

Reply:
I usually prefer boolean flags (almost all others are boolean). What other type can it be?

Reply:
may something related to graphql :man_shrugging: